packages/flutter/lib/src/services/binding.dart

```diff
@@ -82,6 +82,14 @@ mixin ServicesBinding on BindingBase, SchedulerBinding {
   BinaryMessenger get defaultBinaryMessenger => _defaultBinaryMessenger;
   late final BinaryMessenger _defaultBinaryMessenger;
 
+  /// A token that represents the root isolate, used for coordinating with background
+  /// isolates.
+  ///
+  /// This property is primarily intended for use with
+  /// [BackgroundIsolateBinaryMessenger.ensureInitialized], which takes a
+  /// [RootIsolateToken] as its argument.
+  ui.RootIsolateToken? get rootIsolateToken => ui.RootIsolateToken.instance;
```

Comment:
Consider explaining that this will return null if you're not currently on the root isolate.

Also, consider _not_ exposing this at all in the framework's API. Maybe a nicer name for this would be something like `ui.RootIsolateData.token` and `bool get ui.RootIsolteData.isRootIsolate => token != null` or something.

Reply:
I've reworked this and added a `isRootIsolate` to fix mocks for tests.  Everything in `dart:ui` is technically available to users of the framework, so I can't really hide it.  I'm still chasing down making all the web tests working and mocked bindings functions correctly.  Once I have all the lights green I can come back and make sure we are happy with it.